### PR TITLE
MEED-168 Add Filter my spaces by favorite to favorite list page

### PR DESCRIPTION
**Pull request by SaraBoutej**
This change adds new filter to the spaces list toolbar page which allow to filter the list of spaces by favorite space.

### CODE REVIEW

File: component/api/src/main/java/org/exoplatform/social/core/storage/api/SpaceStorage.java [hunk 1]
```diff
@@ -200,6 +200,28 @@ public interface SpaceStorage {
...
+  
+  /**
+   * Gets the count of the favorite spaces of the user id.
+   *
+   * @param userId
+   * @param spaceFilter
+   * @return
+   * @since 1.2.0-GA
+   */
+  public int getFavoriteSpacesByFilterCount(String userId, SpaceFilter spaceFilter);
```
Comment (boubaker) [inline component/api/src/main/java/org/exoplatform/social/core/storage/api/SpaceStorage.java:212]:
```suggestion
  public int countFavoriteSpacesByFilter(String userId, SpaceFilter spaceFilter);
```

**Review by boubaker — COMMENTED** [review]
please add some Unit Tests for new API

**Comment by SaraBoutej** [comment]
> please add some Unit Tests for new API

done

File: component/core/src/test/java/org/exoplatform/social/core/space/spi/SpaceServiceTest.java [hunk 2]
```diff
@@ -450,6 +450,52 @@ public void testGetBookmarkedSpace() throws Exception {
     assertEquals(0, spacesList.length);
     assertEquals(0, spaceService.getAccessibleSpacesByFilter(john.getRemoteId(), spaceFilter).getSize());
   }
+  
+  /**
+   * Test {@link SpaceService#getMemberSpacesByFilter(String, SpaceFilter)}
```
Comment (boubaker) [inline component/core/src/test/java/org/exoplatform/social/core/space/spi/SpaceServiceTest.java:455]:
```suggestion
   * Test {@link SpaceService#getFavoriteSpacesByFilter(String, SpaceFilter)}
```

File: component/core/src/test/java/org/exoplatform/social/core/space/spi/SpaceServiceTest.java [hunk 3]
```diff
@@ -450,6 +450,52 @@ public void testGetBookmarkedSpace() throws Exception {
...
+    createSpace("Space4", john.getRemoteId());
+
+    FavoriteService favoriteService = ExoContainerContext.getService(FavoriteService.class);
+    Favorite space1Favorite = new Favorite(Space.DEFAULT_SPACE_METADATA_OBJECT_TYPE, space1.getId(), null, Long.parseLong(john.getId()));
+    favoriteService.createFavorite(space1Favorite);
+    Favorite space2Favorite = new Favorite(Space.DEFAULT_SPACE_METADATA_OBJECT_TYPE, space3.getId(), null, Long.parseLong(john.getId()));
+    favoriteService.createFavorite(space2Favorite);
+
+    SpaceFilter spaceFilter = new SpaceFilter();
+    assertEquals(4, spaceService.getAccessibleSpacesByFilter(john.getRemoteId(), spaceFilter).getSize());
```
Comment (boubaker) [inline component/core/src/test/java/org/exoplatform/social/core/space/spi/SpaceServiceTest.java:473]:
```suggestion
    assertEquals(2, spaceService.getAccessibleSpacesByFilter(john.getRemoteId(), spaceFilter).getSize());
    assertEquals(2, spaceService.getFavoriteSpacesByFilter(john.getRemoteId(), spaceFilter).getSize());
```

**Review by boubaker — APPROVED** [review]

**Pull request merged**